ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -1371,7 +1373,20 @@ private void prepareForSending() {
             passedInBlobProperties.getContainerId(), passedInBlobProperties.isEncrypted(),
             passedInBlobProperties.getExternalAssetTag(), passedInBlobProperties.getContentEncoding(),
             passedInBlobProperties.getFilename(), resolveReservedMetadataId());
-        operationTracker = getOperationTracker();
+
+        int findValidAttempts = 0;
+        while (findValidAttempts < routerConfig.routerMaxSlippedPutAttempts) {
+          try {
+            // Attempt to get the operation tracker
+            operationTracker = getOperationTracker();
+            break;
+          } catch (Exception e) {
+            findValidAttempts++;
+          }
+        }
```

Comment:
we need this while loop to cover partition selection as well, otherwise, it would just be the same partition and failed over and over again.